File: app/src/main/java/com/github/nikololoshka/pepegaschedule/schedule/model/pair/AttributePair.java
package com.github.nikololoshka.pepegaschedule.schedule.model.pair;

import android.os.Parcelable;

import org.json.JSONException;
import org.json.JSONObject;

public abstract class AttributePair implements Parcelable {
    public abstract void load(JSONObject loadObject) throws JSONException;
    public abstract void save(JSONObject saveObject) throws JSONException;
    public abstract boolean isValid();
}

File: app/src/main/java/com/github/nikololoshka/pepegaschedule/schedule/pair_kotlin/AttributePair.kt
package com.github.nikololoshka.pepegaschedule.schedule.pair_kotlin

import org.json.JSONException
import org.json.JSONObject

/**
 * Базовый абстрактный класс для всех атрибутов пары.
 */
abstract class AttributePair {

    /**
     * Загружает атрибут из json файла.
     * @param jsonObject json объект с парой.
     */
    @Throws(JSONException::class)
    abstract fun load(jsonObject: JSONObject)

    /**
     * Сохраняет атрибут в json файл.
     * @param jsonObject json объект с парой.
     */
    @Throws(JSONException::class)
    abstract fun save(jsonObject: JSONObject)
}